Adds support for Pioneer AVR interface port number

ehagan opened the pull request:
**Description:**
Adds a configuration option for the port the AVR listens on. Some devices do not listen on the default port 23 and listen to 8102 instead.

**Related issue (if applicable):** fixes #
https://community.home-assistant.io/t/support-for-pioneer-avr/503
https://community.home-assistant.io/t/pioneer-av-not-working/2393

**Pull request in [home-assistant.io](https://github.com/home-assistant/home-assistant.io) with documentation (if applicable):** home-assistant/home-assistant.io#1246

**Example entry for `configuration.yaml` (if applicable):**

``` yaml
# Example configuration.yaml entry
media_player:
  - platform: pioneer
    host: 192.168.0.10
    port: 23
```

**Checklist:**

If user exposed functionality or configuration variables are added/changed:
- [x] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If the code communicates with devices, web services, or third-party tools:
- [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
- [ n/a] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
- [n/a] New dependencies are only imported inside functions that use them ([example][ex-import]).
- [n/a] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
- [n/a] New files were added to `.coveragerc`.

https://community.home-assistant.io/t/support-for-pioneer-avr/503
telnetlib supports a port number so adding port as
an optional config element with a default of 23 resolves this.

ehagan committed:
Adds support for Pioneer AVR interface port number

https://community.home-assistant.io/t/support-for-pioneer-avr/503
telnetlib supports a port number so adding port as
an optional config element with a default of 23 resolves this.

ehagan commented:
https://github.com/home-assistant/home-assistant.github.io/pull/1246

fabaff commented on homeassistant/components/media_player/pioneer.py:
```diff
@@ -76,15 +81,15 @@ def telnet_request(cls, telnet, command, expected_prefix):
     def telnet_command(self, command):
         """Establish a telnet connection and sends `command`."""
-        telnet = telnetlib.Telnet(self._host)
+        telnet = telnetlib.Telnet(self._host, self._port)
```
How about adding a timeout as well?

ehagan replied:
@fabaff I can do that.

ehagan committed:
Adds timeout to Pioneer AVR

timeout in telnetlib defaults to socket._GLOBAL_DEFAULT_TIMEOUT
which is not a value, but rather a bare Object used for comparison.

telnetlib says the following about the timeout optional argument:
"The optional timeout parameter specifies a timeout in seconds
 for blocking operations like the connection attempt (if not
 specified, the global default timeout setting will be used)."

From the documentation for sockets:
"Sockets are by default always created in blocking mode"

Catching connect and timeout errors, logging to debug
and continuing.

ehagan committed:
Catches timeout exceptions, logs and continues.

ehagan commented:
The telnetlib Telnet timeout argument is socket level. The default is socket._GLOBAL_DEFAULT_TIMEOUT which is an Object() used for comparison, not a value in seconds. The default value is to block, setting a timeout of 0.0 is non-blocking and a timeout > 0 is blocking for `timeout` seconds. https://docs.python.org/3/library/socket.html#socket-timeouts

It's probably better to have this configurable, especially since the default blocks, but I'm concerned with exposing something this low-level via the config.

fabaff commented:
At some point we should make all timeouts configurable but for now I think that with a 10 s timeout we are OK. In a local net it's pretty safe to say that if a device is not responding within 10 s then it's offline. We are doing this for most parts of `requests`.

fabaff commented:
Can you please fix the conflict? Then we are good to go. Thanks.
